src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -98,79 +149,242 @@ public IndexModel createServerlessIndex(String indexName, String metric, int dim
-
         return createPodsIndex(indexName, dimension, environment, podType, null, null, null, pods, null, null);
     }
 
-    // Minimal + pods + metadata
+    /**
+     * Overload for creating a new pod index with environment, podType, pods, and metadataConfig.
+     *
+     * <pre>
+     *     import io.pinecone.clients.Pinecone;
+     *     import org.openapitools.client.model.CreateIndexRequestSpecPodMetadataConfig;
+     *
+     *     Pinecone client = new Pinecone.Builder("your-api-key").build();
+     *
+     *     CreateIndexRequestSpecPodMetadataConfig metadataConfig = new CreateIndexRequestSpecPodMetadataConfig().fields(Arrays.asList("genre", "year"));
+     *     client.createPodsIndex("my-index", 1536, "us-east4-gcp", "p1.x2", "euclidean", 6);
+     * </pre>
+     *
+     * @param indexName the name of the index to be created.
+     * @param dimension the number of dimensions for the index.
+     * @param environment the cloud environment where you want the index to be hosted.
+     * @param podType the type of pod to use. A string with one of s1, p1, or p2 appended with a "." and one of x1, x2, x4, or x8.
+     * @param pods the number of pods to be used in the index.
+     * @return {@link IndexModel} representing the created serverless index.
+     * @throws PineconeException if the API encounters an error during index creation or if any of the arguments are invalid.
+     */
     public IndexModel createPodsIndex(String indexName, Integer dimension, String environment,
                                       String podType, Integer pods,
                                       CreateIndexRequestSpecPodMetadataConfig metadataConfig) {
-        validatePodIndexParams(indexName, dimension, environment, podType, null, null, null, pods);
-
         return createPodsIndex(indexName, dimension, environment, podType, null, null, null, pods, metadataConfig,
                 null);
     }
 
-    // Minimal + replicas, shards
+    /**
+     * Overload for creating a new pod index with environment, podType, replicas, and shards.
+     *
+     * <pre>
+     *     import io.pinecone.clients.Pinecone;
+     *
+     *     Pinecone client = new Pinecone.Builder("your-api-key").build();
+     *
+     *     client.createPodsIndex("my-index", 1536, "us-east4-gcp", "p1.x2", "euclidean", 2, 2);
+     * </pre>
+     *
+     * @param indexName the name of the index to be created.
+     * @param dimension the number of dimensions for the index.
+     * @param environment the cloud environment where you want the index to be hosted.
+     * @param podType the type of pod to use. A string with one of s1, p1, or p2 appended with a "." and one of x1, x2, x4, or x8.
+     * @param replicas the number of replicas. Replicas duplicate your index. They provide higher availability and throughput and can be scaled.
+     * @param shards the number of shards. Shards split your data across multiple pods so you can fit more data into an index.
```

Comment:
Super nice explanations! I'd add that `replicas*shards == pods` just to be super clear.

Reply:
Same as the last one, I think since the base function where they can pass all 3 calls this out, it's not worth adding in here. This function only exposes `shards` and `replicas` so they don't need to think about `pods` in this case.